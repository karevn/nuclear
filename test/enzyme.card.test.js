import React from 'react';
import { shallow } from 'enzyme';
import Img from 'react-image-smooth-loading';

import Card from '../app/components/Card';


describe('<Card />', () => {
  it('Should render given image', () => {
    const wrapper = shallow(<Card image='IMG' />);
    expect(wrapper.find(Img).prop('src')).toBe('IMG');
  });

  it('Check onClick event', () => {
    const spy = jest.fn();
    const wrapper = shallow(<Card onClick={spy} />);

    wrapper.find('div[onClick]').simulate('click');
    expect(spy).toHaveBeenCalled();
  });

  it('Should display given header', () => {
    const wrapper = shallow(<Card header='HEADER' />);

    expect(wrapper.find('h4').text()).toEqual('HEADER');
  });

  it('Should display given content', () => {
    const wrapper = shallow(<Card content='TEXT' />);

    expect(wrapper.find('p').text()).toBe('TEXT');
  });
});
