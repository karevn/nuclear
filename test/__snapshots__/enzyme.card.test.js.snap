// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Check onClick event 1`] = `
<div
  className="card_container"
>
  <div
    className="card"
    onClick={[MockFunction]}
  >
    <div
      className="thumbnail"
    >
      <Img
        src=""
      />
    </div>
    <div
      className="container"
    >
      <h4 />
    </div>
  </div>
</div>
`;

exports[`<Card /> Should display given content 1`] = `
<div
  className="card_container"
>
  <div
    className="card"
  >
    <div
      className="thumbnail"
    >
      <Img
        src=""
      />
    </div>
    <div
      className="container"
    >
      <h4 />
      <p>
        TEXT
      </p>
    </div>
  </div>
</div>
`;

exports[`<Card /> Should display given header 1`] = `
<div
  className="card_container"
>
  <div
    className="card"
  >
    <div
      className="thumbnail"
    >
      <Img
        src=""
      />
    </div>
    <div
      className="container"
    >
      <h4>
        HEADER
      </h4>
    </div>
  </div>
</div>
`;

exports[`<Card /> Should render given image 1`] = `
<div
  className="card_container"
>
  <div
    className="card"
  >
    <div
      className="thumbnail"
    >
      <Img
        src="IMG"
      />
    </div>
    <div
      className="container"
    >
      <h4 />
    </div>
  </div>
</div>
`;
